### ISSUE TITLE
Add legends to cetz.plot

### ISSUE THREAD

**Issue by Kreijstal** [johannes-wolf/cetz#354]
Just like matplotlib! Teachers will give you negative points if your plots don't have legends.

**Comment by johannes-wolf** [comment]
It's already on 0.2.0.

```typst
#cetz.canvas({
  import cetz.draw: *

  plot.plot(size: (8,6), {
    plot.add(domain: (0, 2*calc.pi), calc.sin, label: $ sin x $)
    plot.add(domain: (0, 2*calc.pi), calc.cos, label: $ cos x $)
  })
})
```

![grafik](https://github.com/johannes-wolf/cetz/assets/519002/151c67ca-c180-482d-966b-6c07a8fd9a10)

**Comment by Kreijstal** [comment]
I guess I'll have to wait till next release in 
https://github.com/typst/packages/

**Comment by dscso** [comment]
I tried this code in 2.0 unfortunately the grid is broken on the x side. Maybe this comment helps as a bug report
```typst
canvas({
    import draw: *

    plot.plot(
        size: (8,6),
        x-grid: true,
        y-grid: true,
        {
            plot.add(domain: (0, 2*calc.pi), calc.sin, label: $ sin x $)
            plot.add(domain: (0, 2*calc.pi), calc.cos, label: $ cos x $)
        }
    )
})
```
<img width="384" alt="image" src="https://github.com/johannes-wolf/cetz/assets/6866008/7f7116e7-4459-4ea1-ae32-ae9152a890f2">

**Comment by johannes-wolf** [comment]
That also got fixed but not yet merged. The fix should be in the bar-/columnchart PR: #337.
